src/commands/es/migrate.ts

```diff
@@ -7,129 +7,151 @@ import emoji from 'node-emoji'
 import { Kommand } from '../../common'
 
 export default class EsMigrate extends Kommand {
-    static initSdk = false
-
-    static description = 'Migrate all the index from one Elasticsearch server to another'
-
-    static flags = {
-        help: flags.help(),
-        src: flags.string({
-            description: 'Source Elasticsearch server URL',
-            env: 'KUZZLE_ES_SRC',
-            required: true
-        }),
-        dest: flags.string({
-            description: 'Destination Elasticsearch server URL',
-            env: 'KUZZLE_ES_DEST',
-            required: true
-        }),
-        reset: flags.boolean({
-            description: 'Reset destination Elasticsearch server',
-            default: false,
-        }),
-        'batch-size': flags.integer({
-            description: 'How many documents to move in batch per operation',
-            default: 1000,
-        }),
-        'no-interactive': flags.boolean({
-            description: 'Skip confirmation interactive prompts (perfect for scripting)',
-            default: false,
-        }),
+  static initSdk = false
+
+  static description = 'Migrate all the index from one Elasticsearch server to another'
+
+  static flags = {
+    help: flags.help(),
+    src: flags.string({
+      description: 'Source Elasticsearch server URL',
+      env: 'KUZZLE_ES_SRC',
+      required: true
+    }),
+    dest: flags.string({
+      description: 'Destination Elasticsearch server URL',
+      env: 'KUZZLE_ES_DEST',
+      required: true
+    }),
+    reset: flags.boolean({
+      description: 'Reset destination Elasticsearch server',
+      default: false,
+    }),
+    'batch-size': flags.integer({
+      description: 'How many documents to move in batch per operation',
+      default: 1000,
+    }),
+    'no-interactive': flags.boolean({
+      description: 'Skip confirmation interactive prompts (perfect for scripting)',
+      default: false,
+    }),
+    'dry-run': flags.boolean({
+      description: 'Print witch collections will be migrated',
+      default: false,
+    }),
+    'pattern': flags.string({
+      description: 'Pattern to match indices to migrate'
+    }),
+    'scroll': flags.string({
+      description: 'Scroll duration for Elasticsearch scrolling',
+      default: '30s'
+    }),
+  }
+
+  static examples = [
+    'kourou es:migrate --src http://elasticsearch:9200 --dest http://otherElasticsearch:9200 --reset --batch-size 2000',
+    'kourou es:migrate --src http://elasticsearch:9200 --dest http://otherElasticsearch:9200 --reset --batch-size 2000 --no-interactive'
+  ]
+
+  private async migrateMappings(index: string, src: Client, dest: Client) {
+    const { body } = await src.indices.getMapping({ index })
+    const mappings = body[index]
+    await dest.indices.create({ index, body: { mappings: mappings.mappings, settings: { number_of_shards: 1, number_of_replicas: 0 } } })
```

Comment:
Why hardcoded `number_of_shards` and replicas?

Reply:
Oh, my bad this was for a one shot migration, i'll make this a flag too

Reply:
I think it should rather be the former values. Otherwise if you want to make it configurable then I think a config file is more appropriate since you may have many indices